### ISSUE TITLE
__init__.py error

### ISSUE THREAD

**Issue by dkssud715** [koulanurag/ma-gym#49]
[/content/ma-gym/ma_gym/__init__.py](https://localhost:8080/#) in <module>
      8 # Register openai's environments as multi agent
      9 # This should be done before registering new environments
---> 10 env_specs = [env_spec for env_spec in envs.registry.all() if 'gym.envs' in env_spec.entry_point]
     11 for spec in env_specs:
     12     register(

AttributeError: 'dict' object has no attribute 'all'

**Comment by koulanurag** [comment]
Can you share you `ma-gym` installation steps? Also, I would recommend to re-try in a [virtualenv](https://docs.python.org/3/library/venv.html).

Also, this [colab link](https://colab.research.google.com/github/koulanurag/ma-gym/blob/master/examples/colab_example.ipynb#scrollTo=Mgxn1Fw6ZAs7) that may help you.

**Comment by 370025263** [comment]
same issue with me.

(pytorch_env) [root@worker01 MARL]# python version.py 
Python版本: 3.6.12 |Anaconda, Inc.| (default, Sep  8 2020, 23:10:56) 
[GCC 7.3.0]
pip版本: 20.2.4
setuptools版本: 50.3.0.post20201006

**Comment by 370025263** [comment]
the problem solved after i changed from gym 0.26 to gym 0.18.3 
`
pip install gym==0.18.3
`
everthing works fine and the problem no longer appears.

GPT answer:

> 
> According to the reference you provided, ma-gym is a collection of multi-agent environments based on OpenAI Gym. It was published in 2019.
> 
> Since 2019, OpenAI Gym has released several versions. The latest stable version as of my knowledge cutoff date is gym 0.18.3. Therefore, it would be appropriate to install gym version 0.18.3.
> 
> You can install gym version 0.18.3 using the following command:
> 
> ```shell
> pip install gym==0.18.3
> ```
> 
> This will ensure that the gym version you install is compatible with ma-gym. Please note that if you use a newer version of gym, there may be compatibility issues with ma-gym.

**Comment by koulanurag** [comment]
Thanks for bringing this to my notice. 

Your previous gym version`( 0.26.2)` is not supported by ma-gym. It will be helpful if you can share your ma-gym installation steps as it will  install `gym>=0.19.0,<=0.20.0` . This can be found in [setup.py file]( https://github.com/koulanurag/ma-gym/blob/master/setup.py#L29C11-L29C33).

For installation, I recommend running following in a new virtual environment.

```
pip install --root-user-action=ignore 'pip<=23.0.1'
pip install --root-user-action=ignore 'setuptools<=66'
pip install --root-user-action=ignore 'wheel<=0.38.4'
pip install --root-user-action=ignore ma-gym
```
